executor/src/wasm_executor.rs

```diff
@@ -141,47 +157,63 @@ impl_function_executor!(this: FunctionExecutor<'e, E>,
 		this.ext.chain_id()
 	},
 	ext_twox_128(data: *const u8, len: u32, out: *mut u8) => {
-		let result =
-			if let Ok(value) = this.memory.get(data, len as usize) {
-				twox_128(&value)
-			} else {
-				[0; 16]
-			};
+		let maybe_value = if len == 0 {
+			Ok(vec![])
+		} else {
+			this.memory.get(data, len as usize)
+		};
+		let result = if let Ok(value) = maybe_value {
+			twox_128(&value)
+		} else {
+			[0; 16]
+		};
 		let _ = this.memory.set(out, &result);
 	},
 	ext_twox_256(data: *const u8, len: u32, out: *mut u8) => {
-		let result =
-			if let Ok(value) = this.memory.get(data, len as usize) {
-				twox_256(&value)
-			} else {
-				[0; 32]
-			};
+		let maybe_value = if len == 0 {
+			Ok(vec![])
+		} else {
+			this.memory.get(data, len as usize)
+		};
+		let result = if let Ok(value) = maybe_value {
+			twox_256(&value)
+		} else {
+			[0; 32]
+		};
 		let _ = this.memory.set(out, &result);
 	},
 	ext_blake2_256(data: *const u8, len: u32, out: *mut u8) => {
-		let result =
-			if let Ok(value) = this.memory.get(data, len as usize) {
-				blake2_256(&value)
-			} else {
-				[0; 32]
-			};
+		let maybe_value = if len == 0 {
+			Ok(vec![])
+		} else {
+			this.memory.get(data, len as usize)
```

Comment:
Would it be possible to get rid of allocation here eventually?

Reply:
Interesting question. I think, probably yes, we can provide memory views (even mutable ones) into wasm memory.

Reply:
yeah - was going to make sure that made it into the new API.